src/powerquery-language-services/library/librarySymbolUtils.ts

```diff
@@ -61,113 +77,134 @@ export function createLibrary(
 
     const completionItemKind: CompletionItemKind | undefined = numberToCompletionItemKind(
         librarySymbol.completionItemKind,
     );
 
-    if (primitiveType === undefined || completionItemKind === undefined) {
-        return undefined;
+    if (primitiveType === undefined) {
+        return failedConversionError(librarySymbol, FailedLibrarySymbolConversionKind.PrimitiveType);
+    }
+
+    if (completionItemKind === undefined) {
+        return failedConversionError(librarySymbol, FailedLibrarySymbolConversionKind.CompletionItemKind);
     }
 
     const label: string = librarySymbol.name;
     const description: string = librarySymbol.documentation?.description ?? "No description available";
 
     if (primitiveType.kind === Type.TypeKind.Type) {
-        return {
+        return ResultUtils.ok({
             kind: Library.LibraryDefinitionKind.Type,
             label,
             description,
             asPowerQueryType: primitiveType,
             completionItemKind,
-        };
+        });
     } else if (librarySymbol.functionParameters) {
         const asPowerQueryType: Type.DefinedFunction | undefined = librarySymbolFunctionSignatureToType(
             librarySymbol,
             primitiveType,
         );
 
         if (asPowerQueryType === undefined) {
-            return undefined;
+            return failedConversionError(librarySymbol, FailedLibrarySymbolConversionKind.AsPowerQueryType);
         }
 
         const parameters: Library.LibraryParameter[] = [];
 
-        for (const parameter of librarySymbol.functionParameters) {
+        for (const [parameter, index] of ArrayUtils.enumerate(librarySymbol.functionParameters)) {
             const libraryParameter: Library.LibraryParameter | undefined =
                 librarySymbolFunctionParameterToLibraryParameter(parameter);
 
             if (libraryParameter === undefined) {
-                return undefined;
+                return failedParameterConversionError(librarySymbol, index);
             }
 
             parameters.push(libraryParameter);
         }
 
-        return {
+        return ResultUtils.ok({
             kind: Library.LibraryDefinitionKind.Function,
             label: TypeUtils.nameOf(asPowerQueryType, NoOpTraceManagerInstance, undefined),
             description,
             asPowerQueryType,
             completionItemKind,
             parameters,
-        };
+        });
     } else {
-        return {
+        return ResultUtils.ok({
             kind: Library.LibraryDefinitionKind.Constant,
             label,
             description,
             asPowerQueryType: primitiveType,
             completionItemKind,
-        };
+        });
     }
 }
 
+function failedConversionError(
+    libarySymbol: LibrarySymbol,
+    kind: FailedLibrarySymbolConversionKind,
+): ErrorResult<FailedLibrarySymbolConversion> {
+    return ResultUtils.error({
+        librarySymbol: libarySymbol,
```

Comment:
non-blocking, but can just be `librarySymbol`. same as below

Reply:
Ah. There was a typo between the two variable names which is why the linter didn't catch this. Updated.